ui/app/src/Utils/FormValidation.elm

```diff
@@ -1,60 +1,57 @@
 module Utils.FormValidation
     exposing
-        ( validateDate
-        , ValidatedMatcher
-        , validatedMatcherToMatcher
-        , validatedMatchersToMatchers
+        ( initialField
+        , ValidationState(..)
+        , ValidatedField
+        , validate
+        , fromResult
+        , stringNotEmpty
+        , updateValue
         )
 
-import Utils.Date exposing (timeFromString)
-import Utils.Types exposing (Matcher)
-import Time exposing (Time)
-import Tuple exposing (second)
 
+type ValidationState
+    = Initial
+    | Valid
+    | Invalid String
 
-validateDate : String -> Result ( String, String ) ( String, Time )
-validateDate dateString =
-    let
-        parsedDate =
-            timeFromString dateString
-    in
-        case parsedDate of
-            Ok date ->
-                Ok ( dateString, date )
 
-            Err err ->
-                Err ( dateString, err )
+fromResult : Result String a -> ValidationState
+fromResult result =
+    case result of
+        Ok _ ->
+            Valid
 
+        Err str ->
+            Invalid str
 
-type alias ValidatedMatcher =
-    { name : Result ( String, String ) String, value : Result ( String, String ) String, isRegex : Bool }
 
+type alias ValidatedField =
+    { value : String
+    , validationState : ValidationState
+    }
 
-validatedMatchersToMatchers : List ValidatedMatcher -> Result String (List Matcher)
-validatedMatchersToMatchers matchers =
-    fold matchers (Ok [])
 
+initialField : String -> ValidatedField
+initialField value =
+    { value = value
+    , validationState = Initial
+    }
 
-fold : List ValidatedMatcher -> Result String (List Matcher) -> Result String (List Matcher)
-fold matchers agg =
-    case matchers of
-        [] ->
-            agg
 
-        m :: list ->
-            fold list (aggregateValidatedMatchers agg m)
+updateValue : String -> ValidatedField -> ValidatedField
+updateValue value field =
+    { field | value = value, validationState = Initial }
 
 
-aggregateValidatedMatchers : Result String (List Matcher) -> ValidatedMatcher -> Result String (List Matcher)
-aggregateValidatedMatchers agg matcher =
-    Result.andThen
-        (\matchers ->
-            Result.map (\m -> m :: matchers) (validatedMatcherToMatcher matcher)
-        )
-        agg
+validate : (String -> Result String a) -> ValidatedField -> ValidatedField
+validate validate field =
+    { field | validationState = fromResult (validate field.value) }
 
 
-validatedMatcherToMatcher : ValidatedMatcher -> Result String Matcher
-validatedMatcherToMatcher matcher =
-    Result.map2 (\n v -> Matcher n v matcher.isRegex) matcher.name matcher.value
-        |> Result.mapError second
+stringNotEmpty : String -> Result String String
```

Comment:
In the long run we should trim the string as well so  a space is not valid either.